bindings/java/src/main/java/org/openucx/jucx/ucp/UcpEndpoint.java

```diff
@@ -8,8 +8,13 @@
 
 import java.io.Closeable;
 import java.nio.ByteBuffer;
+import java.net.InetSocketAddress;
 
 public class UcpEndpoint extends UcxNativeStruct implements Closeable {
+
+    private InetSocketAddress localAddress;
+    private InetSocketAddress remoteAddress;
```

Comment:
why need to save them on the endpoint?

Reply:
It is not necessary to save them, but this way, they only need to be queried once and not every time `getLocalAddress()` or `getRemoteAddress()` are called. The same is done in `UcpListener` and I wanted to keep my implementation consistent with the rest of the code.